apiserver/facades/controller/secretbackendmanager/backends.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/juju/juju/secrets/provider"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/watcher"
+)
+
+var logger = loggo.GetLogger("juju.apiserver.secretbackendmanager")
+
+// SecretBackendsManagerAPI is the implementation for the SecretsManager facade.
+type SecretBackendsManagerAPI struct {
+	resources facade.Resources
+
+	controllerUUID string
+	modelUUID      string
+	modelName      string
+
+	backendRotate BackendRotate
+	backendState  BackendState
+	clock         clock.Clock
+}
+
+// WatchSecretBackendsRotateChanges sets up a watcher to notify of changes to secret backend rotations.
+func (s *SecretBackendsManagerAPI) WatchSecretBackendsRotateChanges() (params.SecretBackendRotateWatchResult, error) {
+	result := params.SecretBackendRotateWatchResult{}
+	w, err := s.backendRotate.WatchSecretBackendRotationChanges()
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	if backendChanges, ok := <-w.Changes(); ok {
+		changes := make([]params.SecretBackendRotateChange, len(backendChanges))
+		for i, c := range backendChanges {
+			changes[i] = params.SecretBackendRotateChange{
+				ID:              c.ID,
+				Name:            c.Name,
+				NextTriggerTime: c.NextTriggerTime,
+			}
+		}
+		result.WatcherId = s.resources.Register(w)
+		result.Changes = changes
+	} else {
+		err = watcher.EnsureErr(w)
+		result.Error = apiservererrors.ServerError(err)
+	}
+	return result, nil
+}
+
+// For testing.
+var (
+	GetProvider = provider.Provider
+)
+
+// RotateBackendTokens rotates the tokens for the specified backends.
+func (s *SecretBackendsManagerAPI) RotateBackendTokens(args params.RotateSecretBackendArgs) (params.ErrorResults, error) {
+	results := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(args.BackendIDs)),
+	}
+	for i, backendID := range args.BackendIDs {
+		backendInfo, err := s.backendState.GetSecretBackendByID(backendID)
+		if err != nil {
+			results.Results[i] = params.ErrorResult{Error: apiservererrors.ServerError(err)}
+			continue
+		}
+		p, err := GetProvider(backendInfo.BackendType)
+		if err != nil {
+			results.Results[i] = params.ErrorResult{Error: apiservererrors.ServerError(err)}
+			continue
+		}
+		if !provider.HasAuthRefresh(p) {
+			continue
+		}
+
+		if backendInfo.TokenRotateInterval == nil || *backendInfo.TokenRotateInterval == 0 {
+			logger.Warningf("not rotating token for secret backend %q", backendInfo.Name)
+			continue
+		}
+
+		logger.Debugf("refresh token for backend %v", backendInfo.Name)
+		cfg := &provider.ModelBackendConfig{
+			ControllerUUID: s.controllerUUID,
+			ModelUUID:      s.modelUUID,
+			ModelName:      s.modelName,
+			BackendConfig: provider.BackendConfig{
+				BackendType: backendInfo.BackendType,
+				Config:      backendInfo.Config,
+			},
+		}
+
+		var nextRotateTime time.Time
+		auth, err := p.(provider.SupportAuthRefresh).RefreshAuth(cfg, *backendInfo.TokenRotateInterval)
+		if err != nil {
+			logger.Errorf("refreshing auth token: %v", err)
```

Comment:
Was this for debugging purposes?
If not, probably include the model name/UUID and `backendInfo.Name` to make it more useful.

Reply:
Just backend name since backends are per controller